awx/ui/client/lib/models/Credential.js

```diff
@@ -40,22 +45,73 @@ function assignInputGroupValues (inputs) {
     });
 }
 
+function setDependentResources (id) {
+    this.dependentResources = [
+        {
+            model: new ProjectModel(),
+            params: {
+                credential: id
+            }
+        },
+        {
+            model: new JobTemplateModel(),
+            params: {
+                credential: id,
+                ask_credential_on_launch: false
+            }
+        },
+        {
+            model: new InventoryModel(),
+            params: {
+                insights_credential: id
+            }
+        },
+        {
+            model: new InventorySourceModel(),
+            params: {
+                credential: id
+            }
+        }
+    ];
+}
+
 function CredentialModel (method, resource, graft) {
     BaseModel.call(this, 'credentials');
 
     this.Constructor = CredentialModel;
     this.createFormSchema = createFormSchema.bind(this);
     this.assignInputGroupValues = assignInputGroupValues.bind(this);
+    this.setDependentResources = setDependentResources.bind(this);
+    this.label = ModelsStrings.get('labels.CREDENTIAL');
 
     return this.create(method, resource, graft);
 }
 
-function CredentialModelLoader (_BaseModel_) {
+function CredentialModelLoader (
+    _BaseModel_,
+    _ProjectModel_,
+    _JobTemplateModel_,
+    _InventoryModel_,
+    _InventorySourceModel_,
+    _ModelsStrings_
+) {
     BaseModel = _BaseModel_;
+    ProjectModel = _ProjectModel_;
+    JobTemplateModel = _JobTemplateModel_;
+    InventoryModel = _InventoryModel_;
+    InventorySourceModel = _InventorySourceModel_;
+    ModelsStrings = _ModelsStrings_;
```

Comment:
After `ModelStrings` is injected, I'd be tempted to refer to it as just `strings` since there's plenty of context. I'm also wondering if we can omit `Model` from the variable names of the models in use. So, `Project` vs. `ProjectModel` etc.